Application Passwords: Section Redesign

Copons opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Fixes DOTCOM-12880

## Proposed Changes

Redesign the Application Password section in the Security settings, as suggested by design.

In detail:

* Update the explanation for consistency with other form explanations (smaller text, italic, lighter color). It's now hidden when the new password screen is displayed.
* Remove the existing passwords list header and wrap items in `CompactCard` containers, for consistency with the Security Key section in the same page. Also hide the list when adding a new password.
* Replace the `cross` delete icon with `trash` as appropriate. Also add a success notice when deleting a password.
* Remove the unnecessary card-in-a-card for the form and new password containers.
* Add a copy button to the new password. The icon turns into a checkmark on click. The action also displays a success or error notice.
* Align the "Done" button to the right to avoid double-clicks on "Generate Password" accidentally dismissing the new password screen.

| Before | After |
|--------|--------|
| ![01_before](https://github.com/user-attachments/assets/64474717-44d0-4557-9651-e249245f3987) | ![01_after](https://github.com/user-attachments/assets/4873b996-0759-480d-85f4-a4b4b38cacf2) |
| ![02_before](https://github.com/user-attachments/assets/be40eaf6-9af4-4d53-939d-7b4a842e7975) | ![02_after](https://github.com/user-attachments/assets/499c71d2-840a-463b-9b8d-309692e143d5) |
| ![03_before](https://github.com/user-attachments/assets/b1beb4be-fd1d-4fa6-8a86-d4f129e15087) | ![03_after_updated](https://github.com/user-attachments/assets/111fefda-5e68-4a0a-9f1a-a864b313d6a3) |
| ![04_before](https://github.com/user-attachments/assets/9e62e0ce-cc13-4e56-bf0c-f07cc027cd81) | ![04_after](https://github.com/user-attachments/assets/00167936-3d5d-4a74-8229-e84322ff7f40) |
| ![05_before](https://github.com/user-attachments/assets/cdaad571-764b-453f-8c07-589e656e14dd) | ![05_after](https://github.com/user-attachments/assets/a2692c9e-a1bf-4a25-9d26-d6fb7c6ea9df) | 

## Additional Notes for Design

@matt-west @fditrapani

The new password element is a regular `p`, not an input field. For this reason I've added the Copy button as an absolutely positioned element on top of if.

I'm not convinced this is the right thing to do, as on smaller screen the button would cover the password.

I wonder if I should entirely replace the element with a `InputControl` as recommended originally. It would need to be disabled, which I think carries its own a11y concerns. 🤔 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Enter My Profile (`/me`).
* Navigate to Security -> Two-Step Authentication.
* Scroll to the bottom to find the Application Passwords section.
* Play around with it and confirm it looks as advertised and feels good to use.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Copons committed:
Update the Application Password list to be a CompactCard list

Copons committed:
Update form look and feel to be more consistent with other forms in the page

Copons committed:
Add an affordance for copying the password

Copons committed:
Add a success notice when deleting an application password

Copons committed:
Restore section's bottom margin

fditrapani commented:
Nice work @Copons. This is a big improvement from before. 

> The new password element is a regular p, not an input field. For this reason I've added the Copy button as an absolutely positioned element on top of if. I'm not convinced this is the right thing to do, as on smaller screen the button would cover the password.

Agreed! Absolutely positioning things should be done sparingly for this reason. Instead, we can use a flex layout , and maybe a little more markup, so that the centre aligned text fills the space available and the button pushes off to the right:

<img width="1549" alt="image" src="https://github.com/user-attachments/assets/44dfa285-9cd8-402f-a33b-031c67df5410" />

While the text is slightly off centre in the main container, it still looks centred because the button balances things out.

matt-west reviewed (commented):
> I wonder if I should entirely replace the element with a InputControl as recommended originally. It would need to be disabled, which I think carries its own a11y concerns. 🤔

Does `InputControl` support the [readonly attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/readonly)? I think that would take care of accessibility concerns as the input would still be focusable, the user just cannot edit it.

Other than that, these changes look good to me.

Copons committed:
Update the new password display with flex and the Core Button component

Copons commented:
@matt-west @fditrapani I think `InputControl` doesn't support `readonly` (yet? it's an experimental component), so I've updated the new password field using flex as recommended.

I've also refactored it a little bit to use the Core `Button` component and to be more responsive, and I've added a "Copy"/"Copied!" label to the button.

![03_after_updated](https://github.com/user-attachments/assets/111fefda-5e68-4a0a-9f1a-a864b313d6a3)

xavier-lc reviewed (commented):
I think it's a bit confusing to have the message `Note: spaces are ignored.` and then have them on the clipboard when using the copy button.

![image](https://github.com/user-attachments/assets/23c241af-5962-4e0a-9bdc-6eece3db9b7d)
![image](https://github.com/user-attachments/assets/108a63c6-1fe3-4dad-aa03-c902f120926b)

xavier-lc reviewed (commented):
Is it me, or the password text looks too big on mobile?

![image](https://github.com/user-attachments/assets/f7b7787f-911c-4973-ae39-4db4e3536911)

xavier-lc reviewed (commented):
The cancel/generate buttons look misaligned on narrow mobile screens.

![chrome-capture-2025-5-9](https://github.com/user-attachments/assets/e37acd18-fad3-4235-800d-e830902048cf)

xavier-lc reviewed (commented):
Apart from that, code changes look good 👌

Copons committed:
Improve responsive view further

Copons commented:
@xavier-lc Thanks for the review, all great points!

* I've updated the "+ Add" button shortening the label on <480px screens.

<img src="https://github.com/user-attachments/assets/b0d526e1-2a2b-49db-aba7-098da90225dc" width="400">

* I've updated the new password with `font-size: 150%` instead of `200%` on <480px screens.

<img src="https://github.com/user-attachments/assets/74632550-bcdf-4f53-abdb-754afe952d6b" width="400">

* I have *not* touched the copy behaviour to filter out spaces because I don't have anything ready to test the password. 😅 

* I have *not* touched the new password form buttons alignment because it's [directed by the `FormButtonsBar` component](https://github.com/Automattic/wp-calypso/blob/3ae367d1b299860c5f5a292e45bfa84c5d936842/client/components/forms/form-buttons-bar/style.scss#L5), and I don't have a strong opinion about it anyway.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17461279

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Copons for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17461279).
